x-pack/plugins/apm/public/components/app/service_overview/service_overview_dependencies_table/index.tsx

```diff
@@ -157,19 +173,33 @@ export function ServiceOverviewDependenciesTable({ serviceName }: Props) {
         }
       ),
       width: px(unit * 5),
-      render: (_, { impact }) => {
-        return <ImpactBar size="m" value={impact} />;
+      render: (_, { currentPeriod, previousPeriod }) => {
+        return (
+          <EuiFlexGroup gutterSize="xs" direction="column">
+            <EuiFlexItem>
+              <ImpactBar value={currentPeriod.impact} size="m" />
+            </EuiFlexItem>
+            {comparisonEnabled && (
+              <EuiFlexItem>
+                <ImpactBar
+                  value={previousPeriod.impact}
+                  size="s"
+                  color="subdued"
+                />
+              </EuiFlexItem>
+            )}
+          </EuiFlexGroup>
+        );
       },
       sortable: true,
     },
   ];
 
   const { data, status } = useFetcher(
     (callApmApi) => {
-      if (!start || !end) {
+      if (!start || !end || !comparisonStart || !comparisonEnd) {
```

Comment:
does this mean it won't be fetched if comparisons are not enabled/active?

Reply:
no, disabling the comparison feature by unchecking the check box won't cause to comparisonStart and comparisonEnd to be undefined. Thus even when comparison is disabled we still fetch data, we only hide it in the UI.

Reply:
If they are not undefined, what is the value i? Do you mean that we always fetch comparison data even if it's not being used?

Reply:
> Do you mean that we always fetch comparison data even if it's not being used?

yes, the first time I showed the feature I mentioned it and everyone agreed. But if you think that's a huge deal I can of course open a new bug and fix it on another PR.

Reply:
Ah, I've got no recollection there unfortunately. Why do we need to do that? To work around the invalidation issue we've seen for other tables?

Reply:
> Why do we need to do that? To work around the invalidation issue we've seen for other tables?

No, there are no technical issues blocking us to fetch or not fetch data when the user toggles on/off the comparison feature. I'd say the only reason why I did it this way was because it's smoother for the user to turn on/off comparison and only show or hide the comparison data instead of refetching all data. Especially because comparison is enabled by default and there's no way to turn it of for good. But I see that if the user turns it off and sets a huge time range that would definitely improve the API responses if the comparison is not fetched alongside with current data.

Reply:
as agreed I'm going to fix it on a separate PR.